test/contrib/test_autoguide.py

```diff
@@ -65,14 +66,17 @@ def model(data, labels):
                                        model_args=(data, labels),
                                        guide_args=(data, labels))
 
+    @jit
     def body_fn(i, val):
         opt_state_, rng_ = val
         loss, opt_state_, rng_ = svi_update(i, rng_, opt_state_,
                                             model_args=(data, labels),
                                             guide_args=(data, labels))
         return opt_state_, rng_
 
-    opt_state, _ = lax.fori_loop(0, 1000, body_fn, (opt_state, rng_train))
+    # TODO: Remove once https://github.com/pyro-ppl/numpyro/issues/259 is fixed.
```

Comment:
The while loop compilation here takes more than 300 seconds, but this is quite fast if we run this with a native for loop.

Reply:
Are you using dev version of `jax`? In the new version, we either have to jit body_fn or jit fori_loop.

Reply:
No, I'm using the release version. The issue is that `while_loop` triggers an xla compilation by default and that takes forever (more than 300 seconds). However, if we replace that with a python for loop, it takes only 11 seconds to run.